LOG-5637: e2e Cluster Logging OTEL support (#1684)
openshift/enhancements: enhancements/cluster-logging/cluster-logging-otel-support.md

The developer role is any member of the Red Hat OpenShift Logging team.

### Goals

* Provide an in-cluster log search and alerting experience based on [OpenTelemetry Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/)
* Provide an OTLP-native log ingestion for third party trusted/verified collectors.
* Provide native zero-effort OpenShift Logging Tenancy and RBAC integration to third-party trusted/verified collectors
* Provide a transition period for users to continue to use the OpenShift web console until they must modify their logQL to use OTEL

### Non-Goals

* Entirely replace the [ViaQ](https://github.com/openshift/cluster-logging-operator/blob/release-5.9/docs/reference/datamodels/viaq/v1.adoc) data model with OpenTelemetry Semantic Conventions.
* In-cluster log storage for public, non-verified log ingestion.

## Proposal

In order to realize the goals of this enhancement:

* Modify **cluster-logging-operator** to allow writing to **LokiStack** over ViaQ(the default) or OTLP based upon a **ClusterLogForwarder** annotation
* Modify the vector log collector deployment to write to **LokiStack** over OTLP and additionally include select ViaQ attributes

> * Update the **loki-operator** to deploy v3.x are better of Loki to provided the an OTLP endpoint

Author: I'm trying to suggest we need a version of Loki that requires an ingest endpoint.  Is it correct to say the LO will provide an API to configure OTLP?  Won't the version of Loki have it enabled already?  Is it something a user can configure?  I defer to log-storage-team here on best wording

> * Update the **loki-operator** to deploy v3.x are better of Loki to provided the an OTLP endpoint

Reviewer: > Is it correct to say the LO will provide an API to configure OTLP? 

I would say so, the version of Loki that's in 6.0 already supports OTLP but we do not yet support users configuring it through LokiStack

> * Update the **loki-operator** to deploy v3.x are better of Loki to provided the an OTLP endpoint

Author: @JoaoBraveCoding 
* Is the wording sufficient here?  
* Is there work to be done for LokiStack to realize this enhancement?
* Do we need a JIRA to support :arrow_up:

> * Update the **loki-operator** to deploy v3.x are better of Loki to provided the an OTLP endpoint

Reviewer: > Is the wording sufficient here?

From my POV I suggest the wording I made in the suggestion above.

> Is there work to be done for LokiStack to realize this enhancement?

Yes, LokiStack has to be modified to accommodate and allow the user to configure Loki OTLP ingestion. That work has been done in
- https://github.com/grafana/loki/pull/13446

> Do we need a JIRA to support

https://issues.redhat.com/browse/LOG-5673

* managing and deploying observability operands and console plugins (e.g Logging UIPlugin)

**loki-operator** is:

* an operator responsible for managing a LokiStack instance.

**cluster-logging-operator** is:

* an operator responsible for managing log collection and forwarding.

The cluster administrator does the following:

1. Deploys the Red Hat **cluster-observability-operator**
1. Deploys the Red Hat **loki-operator** which supports server-side indexing and OTLP ingestion endoint
1. Deploys an instance of **LokiStack** into the `openshift-logging` namespace
1. Deploys the Red Hat **cluster-logging-operator**
1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
    1. **NOTE:** In Tech Preview, this resources includes an annotation to instruct the operator to use the OTEL datamodel in lieu of ViaQ

The **cluster-observability-operator**:

1. Deploys the console **Logging UIPlugin** for reading logs in the OpenShift console

The **loki-operator**

1. Deploys the **LokiStack** for storing logs in-cluster

The **cluster-logging-operator**:

1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace

### API Extensions

> During the time this feature is considered Tech Preview, instances of **ClusterLogForwarder** will be [annotated](./forwarder-to-otlp.md) to enable

Reviewer: Seems like this reference is broken

>    observability.openshift.io/enable-lokistack-otlp: [[LOKISTACK_NAMES]]

Reviewer 2: Do we need the annotation anymore? AFAIU we have de data model field for this purpose now.

Reviewer 3: As far as I remember we settled on "field for selecting the data model and an annotation to enable the tech-preview feature" on Friday.

forwarding to OTLP receivers. Additionally, a field will be added to the **LokiStack** output type where administrators can configure the desired
data model of forwarded logs.

Example of configuring multiple outputs:

```yaml
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder

> spec:

Reviewer: Should we add the required tech preview annotation here? Unless I missed it in this document we don't specify what the annotation looks like

  - name: my-loki-otel
    type: lokiStack
  - name: my-other-loki-otel
    type: lokiStack
  - name: my-legacy-loki
    type: lokiStack
  pipelines:
  - name: in-to-out
    outputRefs: [my-loki-otel,my-other-loki-otel,my-legacy-loki]
    inputRefs: [application]
```

This example configures a collector to forwards logs to multiple LokiStacks (e.g. my-loki-otel,my-other-loki-otel) using OTEL semantic attributes while
also forwarding logs to a separate LokiStack (e.g. my-legacy-loki) using ViaQ.

### Topology Considerations

#### Hypershift / Hosted Control Planes

#### Standalone Clusters

#### Single-node Deployments or MicroShift

### Implementation Details/Notes/Constraints

#### Log Storage

* The **loki-operator** will be updated to the version which is capable of providing an OTLP endpoint so that Loki deployments can ingest log entries from any OTLP client.  This is required to allow
trusted and verified third-party Red Hat products, like Network Observability, to forward logs to the same LokiStack as the Red Hat OpenShift Logging.  Additionally this version of Loki will

> support server-side indexing to improve query performance.

Author: Tweaking the wording a bit

* The **loki-operator** will be updated so deployments of Lokistack rely upon the OTLP API instead of the Push API. This will allow
trusted and verified third-party Red Hat products, like Red Hat Build of the OpenTelemetry Collector, to forward logs to the same LokiStack as the Red Hat OpenShift Logging.  Additionally, this new version of the **loki-operator**

> will support server-side indexing to improve query performance and simplify configuration on the client side. Sane defaults will be provided for users using the `openshift-*` tenancies.

Reviewer 3: Currently I am under the impression that we're only providing OTLP defaults for logging and not for the network observability operator?

```suggestion
will support server-side indexing to improve query performance and simplify configuration on the client side. Sane defaults will be provided for users using the `openshift-logging` tenancy.
```

Author: > Currently I am under the impression that we're only providing OTLP defaults for logging and not for the network observability operator?

@xperimental who can answer this question and what changes need to be made to the document?

Author: Determined replacing "tenancy" to be only "openshift-logging" addresses this comment and my follow-up question.

will support server-side indexing to improve query performance and simplify configuration on the client side. Sane defaults will be provided for users using the `openshift-logging` tenancy.
* The **loki-operator** will configure Loki deployments with server-side indexing of OTEL and Minimal ViaQ Product Labels (MVP).  The Red Hat Observability Logging 
[Data Model](https://github.com/rhobs/observability-data-model/pull/6)
provides additional details.

#### Log Visualization

> The **Logging UIPlugin** will be modified to replace its UI widgets to rely upon OTEL attributes in lieu of ViaQ.  These changes will be

Reviewer 4: These changes will be released to production one release after the other components achieve **GA**.   can you explain this sentence?  I think Logging visualization must be ready when the other components achieve **GA**.

> The **Logging UIPlugin** will be modified to replace its UI widgets to rely upon OTEL attributes in lieu of ViaQ.  These changes will be

Reviewer 4: These changes will be released to production one release after the other components achieve **GA**.   can you explain this sentence?  I think Logging visualization must be ready when the other components achieve **GA**.

Author: This enhancement provides a path for users to continue to use visualization when there is a mismatch between visualization and what attributes are forwarded by the collector.  The intent, as suggested by PM, is to only forward OTEL attributes when GA, allow old data to essentially expire or role off, and then introduce the visualization that only relies upon OTEL attributes.

Reviewer 4: Got it, thanks

* Modify **cluster-logging-operator** to allow writing to **LokiStack** over ViaQ(the default) or OTLP based upon a **ClusterLogForwarder** annotation
* Modify the vector log collector deployment to write to **LokiStack** over OTLP and additionally include select ViaQ attributes
* Update the **loki-operator** to deploy v3.x are better of Loki to provided the an OTLP endpoint
* Update the **loki-operator** to provide sane defaults to the new LokiStack OTEL API for server-side indexing of OTEL attributes to improve query performance
* Modify the **logging-view-plugin** to use OTEL attributes that correspond to ViaQ *log_type*, *log_source*, *kubernetes.container_name*, *kubernetes.namespace_name*, *kubernetes.pod_name*

### Workflow Description

The following workflow describes deployment of a full logging stack to collect and forward logs to a Red Hat managed log store, **LokiStack**, using OTLP.

**cluster administrator** is a human responsible for:

* Managing and deploying day 2 operators
* Managing and deploying an in-cluster LokiStack
* Managing and deploying a cluster-wide log forwarder

**cluster-observability-operator** is an operator responsible for:

> * managing and deploying observability operands (e.g. LokiStack, ClusterLogForwarder, Tracing) and console plugins (e.g console-logging-plugin)

Reviewer 2: ```suggestion
* managing and deploying observability operands and console plugins (e.g console-logging-plugin)
```

Removing the examples for operands here to avoid confusion. As per now it is not possible to install CLF/LokiStack/Tracing via COO. This is a future outcome and has not materialized yet.

**cluster-logging-operator** is an operator responsible for managing log collection and forwarding.

The cluster administrator does the following:

1. Deploys the Red Hat **cluster-observability-operator**
1. Deploys the Red Hat **loki-operator** which supports server-side indexing and OTLP ingestion endoint
1. Deploys an instance of **LokiStack** in the `openshift-logging` namespace
1. Deploys the Red Hat **cluster-logging-operator**
1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
    1. **In Tech Preview, additionally annotated to instruct the operator to use the OTEL datamodel in lieu of ViaQ**

The **cluster-observability-operator**:

1. Deploys the console-logging-plugin for reading logs in the OpenShift console

The **loki-operator**

1. Deploys the **LokiStack** for storing logs in-cluster

The **cluster-logging-operator**:

1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace

### API Extensions

During the time this feature is considered Tech Preview, instances of **ClusterLogForwarder** will be annotated to indicate to the
operator that logs will be forwarded to **LokiStack** using OTLP:

```

>    logstore.observability.openshift.io/otlp: [[LOKISTACK_NAMES]]

Reviewer 2: I suggest to adapt this w/o reviving the `logstore` term from the old `ClusterLogging` resource:
```
   logging.observability.openshift.io/select-lokistack-otlp: [[LOKISTACK_NAMES]]
```
or
```
   observability.openshift.io/enable-lokistack-otlp: [[LOKISTACK_NAMES]]
```

```
* LOKISTACK_NAMES := Comma delimited list of **clusterlogforwarder.spec.output[].name** of type **lokiStack**

Example of configuring multiple outputs:

```yaml
apiVersion: "observability.openshift.io/v1"
kind: ClusterLogForwarder
metadata:
  annotations:
    logstore.observability.openshift.io/otlp: my-loki-otel,my-other-loki-otel
spec:
  outputs:
  - name: my-loki-otel
    type: lokiStack
  - name: my-other-loki-otel
    type: lokiStack
  - name: my-legacy-loki
    type: lokiStack
  pipelines:
  - name: in-to-out
    outputRefs: [my-loki-otel,my-other-loki-otel,my-legacy-loki]
    inputRefs: [application]
```

This example configures a collector to forwards logs to multiple LokiStacks (e.g. my-loki-otel,my-other-loki-otel) using OTEL semantic attributes while
also forwarding logs to a separate LokiStack (e.g. my-legacy-loki) using ViaQ.

### Topology Considerations

#### Hypershift / Hosted Control Planes

#### Standalone Clusters

#### Single-node Deployments or MicroShift

### Implementation Details/Notes/Constraints

#### Log Storage

> * The **loki-operator** will be updated so deployments of Lokistack rely upon the OTLP API instead of the Push API to the OTLP API. This will allow
> trusted and verified third-party Red Hat products, like Network Observability, to forward logs to the same LokiStack as the Red Hat OpenShift Logging.  Additionally, this new version of the **loki-operator**
> will support server-side indexing to improve query performance and simplify configuration on the client side. Sane defaults will be provided for users using the `openshift-*` tenancies.
> * The **loki-operator** will configure Loki deployments with server-side indexing of OTEL and Minimal ViaQ Product Labels (MVP)

Reviewer 2: ```suggestion
* The **loki-operator** will be updated so deployments of Lokistack rely upon the OTLP API instead of the Push API. This will allow
trusted and verified third-party Red Hat products, like Red Hat Build of the OpenTelemetry Collector, to forward logs to the same LokiStack as the Red Hat OpenShift Logging.  Additionally, this new version of the **loki-operator**
will support server-side indexing to improve query performance and simplify configuration on the client side. Sane defaults will be provided for users using the `openshift-*` tenancies.
* The **loki-operator** will configure Loki deployments with server-side indexing of OTEL and Minimal ViaQ Product Labels (MVP)
```

```

> LIST OF LABLES

Reviewer 2: ```suggestion
spec:
  limits:
    global:
      otlp:
        # Resource Attribtutes always treated as index labels by the distributor
        indexedResourceAttributes:
        # Labels provided by all RH observability log collectors
        - service.name
        - k8s.cluster.uid
        - k8s.namespace.name
        - k8s.pod.name
        - k8s.container.name
        # Labels provided by OpenShift Logging Vector
        - openshift.log.source
        # Labels provided by OpenTelemetryCollector
        - k8s.deployment.name
        - k8s.statefulset.name
        - k8s.daemonset.name
        - k8s.cronjob.name
        # Labels for OpenShift Logging UI backward compatibility
        - log_type
        - kubernetes_namespace_name
        - kubernetes_pod_name
        - kubernetes_container_name
        resourceAttributes:
          ignoreDefaults: false
          attributes:
          - action: structured_metadata
            attributes:
            - k8s.replicaset.name
            - k8s.job.name
        logAttributes:
        # Structured medata provided by OpenShift Logging Vector
        - action: structured_metadata
          attributes:
          # For openshift log source container
          - k8s.node.name
          # For openshift log source node
          - service.name
          - host.name
          - process.command
          # For openshift log source auditd
          - node.name
          # For openshift log sources kubeAPI and openshiftAPI
          - url.path
          - http.request.method_original
          - http.response.status.code
```

---
title: cluster-logging-otel-support
authors:
  - "@jcantrill"
reviewers:
  - "@alanconway, Red Hat Logging Architect"
  - "@periklis, Red Hat Logging team lead"
  - "@xperimental, Red hat Log Storage team member"
  - "@JoaoBraveCoding, Red hat Log Storage team member"
approvers:
  - "@alanconway, Red Hat Logging Architect"
api-approvers:
  - "@alanconway"
  - "@periklis"
creation-date: 2024-09-25
last-updated: 2024-09-25
tracking-link:

>   - [LOG-5637]https://issues.redhat.com/browse/LOG-5637

Author: I changed this in a commit I haven't pushed so I guess I'm ok now

### User Stories

* As an administrator, I want to deploy an in-cluster logging solution that makes use of the OpenTelementry Observability framework
so I can make correlations with other observability signals (i.e. metrics, tracing) in an industry, standardized way
* As a developer, I want users to be able to query log records using OTEL or [ViaQ](https://github.com/openshift/cluster-logging-operator/blob/release-5.9/docs/reference/datamodels/viaq/v1.adoc)
attributes: log type, namespace, pod name, container name to facilitate LokiStack's transition from using ViaQ to OTEL
* As a developer, I want Red Hat LokiStack to provide a native OTLP ingestion point
so that other Red Hat products can forward observability signals using OTEL and rely upon OpenShift Logging tenancy and RBAC integration

The administrator role is any user who has permissions to deploy the operator and the cluster-wide resources required to deploy the logging components.
The developer role is any member of the Red Hat OpenShift Logging team.

### Goals

* Provide an in-cluster log search and alerting experience based on [OpenTelemetry Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/)

> * Provide an OTLP-native log ingestion for third party trusted/verified collectors.

Author: This statement comes from the original Epic and I believe is intended to express a path for the RH build of the OTEL collector and for network observability to write logs into RH Logging log storage.  The additional expectation is those sources provide the MVP labels if they expect the console ui integration to be capable of retrieving those records.  I'm open to suggestions on how to better clarify ad I do not think we expect to alter the gateway to provide validation

> * Provide an OTLP-native log ingestion for third party trusted/verified collectors.

Reviewer 2: This is not meant as technical requirement. More to inform that RH Build of OpenTelemetry Collector adheres to the minimum ViaQ labels as long as we need them in the Logging UI.

> * Provide an OTLP-native log ingestion for third party trusted/verified collectors.

Author: Discussion we decided to remove references to "third party"

1. Deploys the Red Hat **cluster-logging-operator**
1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
    1. **In Tech Preview, additionally annotated to instruct the operator to use the OTEL datamodel in lieu of ViaQ**

The **cluster-observability-operator**:

1. Deploys the console-logging-plugin for reading logs in the OpenShift console

The **loki-operator**

1. Deploys the **LokiStack** for storing logs in-cluster

The **cluster-logging-operator**:

1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace

### API Extensions

During the time this feature is considered Tech Preview, instances of **ClusterLogForwarder** will be annotated to indicate to the
operator that logs will be forwarded to **LokiStack** using OTLP:

```
   observability.openshift.io/enable-lokistack-otlp: [[LOKISTACK_NAMES]]
```
* LOKISTACK_NAMES := Comma delimited list of **clusterlogforwarder.spec.output[].name** of type **lokiStack**

Example of configuring multiple outputs:

```yaml
apiVersion: "observability.openshift.io/v1"
kind: ClusterLogForwarder
metadata:
  annotations:

>     logstore.observability.openshift.io/otlp: my-loki-otel,my-other-loki-otel

Author: > configuration attribute on the output

What does this look like?  The intent of keeping an annotation is to not have to add a field into the API spec which because obsolete once writing to LokiStack only via OTLP

>  once the feature becomes GA but customers still need a "mixed output" they can continue to use the attribute and we just switch the meaning of "no value provided" at some point in the future

I anticipate adding a complimentary annotation when the feature goes GA to allow enablement of ViaQ for lokiStack which goes away GA+1 since that is when the UI is expected to reach parity

>     logstore.observability.openshift.io/otlp: my-loki-otel,my-other-loki-otel

Author: Discussion we resolved:
* Rely only upon TP enablement annotation
* Add field to lokiStack output: `dataModel=(""|viaq|otel)`
* TP: `""=viaq`
* GA: `""=otel`

cc @Clee2691

        # Labels provided by OpenShift Logging Vector
        - openshift.log.source
        # Labels provided by OpenTelemetryCollector
        - k8s.deployment.name
        - k8s.statefulset.name
        - k8s.daemonset.name
        - k8s.cronjob.name
        # Labels for OpenShift Logging UI backward compatibility
        - log_source
        - log_type
        - kubernetes_namespace_name
        - kubernetes_pod_name
        - kubernetes_container_name
        resourceAttributes:
          ignoreDefaults: false
          attributes:
          - action: structured_metadata
            attributes:
            - k8s.replicaset.name
            - k8s.job.name
        logAttributes:
        # Structured medata provided by OpenShift Logging Vector
        - action: structured_metadata
          attributes:
          # For openshift log source container
          - k8s.node.name
          # For openshift log source node
          - service.name
          - host.name
          - process.command
          # For openshift log source auditd
          - node.name
          # For openshift log sources kubeAPI and openshiftAPI
          - url.path
          - http.request.method_original

>           - http.response.status.code

Author: * Are you recommending we remove this list?
* Are you recommending we describe what we intend to do?

>           - http.response.status.code

Author: Removing in favor of referencing https://github.com/rhobs/observability-data-model/pull/6

```

#### Log Visualization

The **log-view-plugin** will be modified to replace its UI widgets to rely upon OTEL attributes in lieu of ViaQ.  (Review meeting notes for additional info)

#### Log Collection and Forwarding

* The **cluster-logging-operator** will be updated to evaluate the OTLP annotation:
    * Absence of the annotation implies forwarding to **LokiStacks** using ViaQ data model

>     * Existance of the annotation, the operator:

Reviewer 3: ```suggestion
    * Existence of the annotation, the operator:
```

        * will configure **LokiStack** OTLP outputs to include MVP labels

Once this enhancement achieves General Acceptance, OTEL will be the default data model when forwarding logs to **LokiStack**.  Cluster administrators will
be able to support the Tech Preview behavior until it is removed by annotating the **ClusterLogForwarder** like the following:

```
   logstore.observability.openshift.io/viaq: [[LOKISTACK_NAMES]]
```
* LOKISTACK_NAMES := Comma delimited list of **clusterlogforwarder.spec.output[].name** of type **lokiStack**

* The **cluster-logging-operator** will be updated to evaluate the ViaQ annotation:
    * Absence of the annotation implies forwarding to **LokiStacks** using OTEL data model
    * Existance of the annotation, the operator:
        * will validate the list of names maps to a defined, **LokiStack** output and rejects the **ClusterLogForwarder** for any mismatches
        * will internally initialize Loki outputs for the enabled **LokiStacks**, in lieu of OTLP outputs
        * will configure **LokiStack** LokiStack outputs to include OTEL equivalent MVP labels

52
### Risks and Mitigations

### Drawbacks

> This design offers a transitional plan to allow users to change the enrichment model of their log storage while contining to use the console web UI plugin and

Reviewer 3: ```suggestion
This design offers a transitional plan to allow users to change the enrichment model of their log storage while continuing to use the console web UI plugin and
```

MVP attributes.  Advanced queries using [logQL](https://grafana.com/docs/loki/latest/query/) that rely upon attributes other then MVP may no longer function.

## Open Questions [optional]

1. Do we need to forward the complimentry UI OTEL attributes when configured for ViaQ?

## Test Plan

## Graduation Criteria

### Dev Preview -> Tech Preview

- Ability to conditionally configure the datamodel(ViaQ or OTEL) when forwarding to in-cluster LokiStack

> - Ability to retrieve OTEL formatted log entries from in-cluster LokiStack using ViaQ attributes: log_type, kubernetes_namespace_name, kubernetes_container_name, kubernetes_pod_name

Reviewer 3: Our current set of labels/attributes is [this](https://github.com/openshift/cluster-logging-operator/blob/master/internal/generator/vector/output/loki/loki.go#L34-L51):

- `log_type`
- `kubernetes.container_name`
- `kubernetes.host`
- `kubernetes.namespace_name`
- `kubernetes.pod_name`

I think these are good candidates for the "required" set, maybe we should add the new OTLP counterparts to this list as well.

Author: adding `log_source` to the list.

Are you suggesting we alter this text to identify both sets side-by-side?

Author: Resolving as document was modified to move these to a dedicated document

- Ability to use existing log based alerts after logging upgrade without changes
- Announce deprecation of log forwarding to LokiStack using ViaQ data model, to be removed **two** logging releases after GA

### Tech Preview -> GA

- Console UI plugin enhanced to replace ViaQ(i.e. log_type, kubernetes_namespace_name, kubernetes_container_name, kubernetes_pod_name) attribute dependencies with OTEL equivalents
- Forwarding to in-cluster LokiStack with OTEL datamodel by default
- Gather feedback from consumers of Red Hat OpenShift Logging
- Official, Red Hat user facing documentation

### Removing a deprecated feature

## Upgrade / Downgrade Strategy

## Version Skew Strategy

## Operational Aspects of API Extensions

## Support Procedures

## Alternatives

> As an alternative to the above proposed solution (support of OpenTelemetry Semantic Conventions formated log records via OTLP in addition to ViaQ) 

Reviewer 3: ```suggestion
As an alternative to the above proposed solution (support of OpenTelemetry Semantic Conventions formatted log records via OTLP in addition to ViaQ)
```

        * will internally initialize Loki outputs for the enabled **LokiStacks**, in lieu of OTLP outputs
        * will configure **LokiStack** LokiStack outputs to include OTEL equivalent MVP labels

52
### Risks and Mitigations

### Drawbacks

This design offers a transitional plan to allow users to change the enrichment model of their log storage while continuing to use the console web UI plugin and
MVP attributes.  Advanced queries using [logQL](https://grafana.com/docs/loki/latest/query/) that rely upon attributes other then MVP may no longer function.

## Open Questions [optional]

1. Do we need to forward the complimentry UI OTEL attributes when configured for ViaQ?

## Test Plan

## Graduation Criteria

### Dev Preview -> Tech Preview

- Ability to conditionally configure the datamodel(ViaQ or OTEL) when forwarding to in-cluster LokiStack
- Ability to retrieve OTEL formatted log entries from in-cluster LokiStack using ViaQ attributes: log_type, kubernetes_namespace_name, kubernetes_container_name, kubernetes_pod_name
- Ability to use existing log based alerts after logging upgrade without changes

> - Announce deprecation of log forwarding to LokiStack using ViaQ data model, to be removed **two** logging releases after GA

Reviewer: With the move to OTLP and away from ViaQ for tenant `openshift-logging` in a sense, we are also deprecating `schemas` < v13 correct? Do we need to "prepare" for this? We already have an alert informing users to update but maybe for 6.1 we should tune the alert, WDYT? cc @periklis @xperimental

Reviewer 3: We probably should update the message on the existing alert, so that it says something like "OTLP will not work correctly before v13". Interestingly, it seems to just ignore the `structured_metadata` when ingested over OTLP with an older schema version instead of returning an error as with the Push-API.

Author: Impl detail from LokiStack not requiring changes to this document. Resolving comment
